src/NuGet.Core/NuGet.Commands/RestoreCommand/VulnerabilityInformationProvider.cs

```diff
@@ -17,19 +17,19 @@ internal sealed class VulnerabilityInformationProvider : IVulnerabilityInformati
     {
         private readonly SourceRepository _source;
         private readonly ILogger _logger;
-
+        private readonly CancellationToken _cancellationToken;
         private readonly AsyncLazy<GetVulnerabilityInfoResult?> _vulnerabilityInfo;
 
-        public VulnerabilityInformationProvider(SourceRepository source, ILogger logger, bool isAuditSource)
+        public VulnerabilityInformationProvider(SourceRepository source, ILogger logger, bool isAuditSource, CancellationToken cancellationToken)
         {
             _source = source;
             _logger = logger;
             IsAuditSource = isAuditSource;
-
+            _cancellationToken = cancellationToken;
             _vulnerabilityInfo = new AsyncLazy<GetVulnerabilityInfoResult?>(GetVulnerabilityInfoAsync);
```

Comment:
Is it possible we do something like
```suggestion
            _vulnerabilityInfo = new AsyncLazy<GetVulnerabilityInfoResult?>(
                () => GetVulnerabilityInfoAsync(CancellationToken.None));
```

Reply:
When the method took an argument, specifying the delegate made sense but when I removed the argument, VS recommended simplifying the lamda.

Reply:
I meant we could do
`_vulnerabilityInfo = new AsyncLazy<GetVulnerabilityInfoResult?>(
                () => GetVulnerabilityInfoAsync(cancellationToken));
`
and not make the _cancelation token part of the class. In addition, put the defintion of `GetVulnerabilityInfoAsync(cancellationToken)` back to one that takes a cancelation token as an argument. 

I was thinking if we make the cancelation token part of the class, it could result in all ongoing tasks stopping regardless of their individual needs. However, I don't think we create multiple calls to `GetVulnerabilityInfoAsync(cancellationToken)` with different canclationTokens so I guess it should not be a problem for this specific scenario.

Reply:
If multiple threads call `GetVulnerabilityInformationAsync(CancellationToken token)`, but the method itself is just calling `GetVulnerabilityInfoAsync()` once and returning a cached value to all the other threads, which `CancellationToken` should be respected?

My thought was the creation of this class happens once during restore and at that time there's only a single `CancellationToken` for the whole restore operation.  Multiple threads are then running to restore individual projects and they call into this provider to get the vulnerability information, but this class only does the operation once. So I'm not sure if this individual method needs to accept a cancellation token, checking vulnerability information is a single operation as part of restore.